File: LeetCode/javascript solutions/Maximum Difference Between Increasing Elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var difference = (x,y) =>
{
    return (x>y) ? x : y;
}

var maximumDifference = (nums)=> {
    let diff = Number.MIN_VALUE;
    if(nums.length ==0)
        return diff;
    
    for( let i=0; i<nums.length -1; i++) {  
        for( let j=i; j< nums.length; j++)
            {
                if(nums[j] > nums [i])    
                    diff = difference(diff, nums[j] - nums [i])
            }
       }
    if(diff<1) return -1;
    return diff;
};

File: LeetCode/Maximum Difference Between Increasing Elements.cpp
class Solution {
public:
    int max(int x, int y) {
    return (x > y) ? x : y;
    }
    int maximumDifference(vector<int>& nums) {
        int diff = INT_MIN;
 
    if (nums.size() == 0) {
        return diff;
    }
 
    for (int i = 0; i < nums.size() - 1; i++) {
        for (int j = i + 1; j < nums.size(); j++) {
            if (nums[j] > nums[i]) {
                diff = max(diff, nums[j] - nums[i]);
            }
        }
    }
    if(diff<-1) return -1;
    return diff;
    }
};
